### PR TITLE
Update license to CC-BY-NC-SA-4.0

### PR DESCRIPTION
**Pull request by presidentbeef**
Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International Public License

### CODE REVIEW

**Comment by presidentbeef** [comment]
To clarify:

[As noted](https://github.com/presidentbeef/brakeman/blob/master/COPYING), we are not "relicensing" existing code committed prior to June 15, 2018. We are sub-licensing as explicitly allowed by the MIT license. All preexisting code is still available under MIT terms.

Anyone may modify and/or redistribute the Brakeman code as allowed under [CC-BY-NC-SA-4.0](https://github.com/presidentbeef/brakeman/blob/master/CC-LICENSE.md).

The only restriction on use or distribution of Brakeman code is when it is included as part of a commercial product *that uses Brakeman as a feature of the product itself* (i.e. ["Brakeman-as-a-Service"](https://github.com/presidentbeef/brakeman/wiki/Brakeman-as-a-Service)). This does not prevent you from using Brakeman to scan your own applications or even from using Brakeman as part of a professional security assessment. It does not mean you cannot include the Brakeman gem in your Gemfile, even if your application is a commercial product.

We will be asking that future contributions be explicitly assigned to Synopsys, Inc. and made available under the same Creative Commons license. However, please note that Brakeman, Inc. also has a irrevocable perpetual license to the Brakeman code owned by Synopsys. This allows us to guarantee the continued availability of Brakeman.

As the author of nearly 90% of Brakeman's current code, the original creator, and the maintainer for the past eight years and 109 releases, I am committed to continuing to maintain Brakeman in the same manner I have been doing so since its inception. It is literally part of my job now.

In other words, the vast majority of Brakeman users are not affected by this license change and should not be worried about the future of Brakeman.

**Comment by chloerei** [comment]
https://brakemanpro.com/2018/06/28/brakeman-pro-acquired-by-synopsys

> However, under the new license, it is no longer possible to use Brakeman OSS for the development of a commercial product or online service **or** to resell Brakeman OSS as a service. 

Is it the 'or' redundant? It's confused me that if I can use Brakeman to scan commercial product that not using Brakeman as a feature.

**Comment by presidentbeef** [comment]
@chloerei I apologize for the wording. Sometimes the difference between lawyers and developers causes confusion on both sides.

**You may absolutely use Brakeman to scan a commercial application.**

**Comment by presidentbeef** [comment]
I have added some additional information in #1239 

- Brakeman can be freely used, modified, or distributed for any purpose except as a feature of a commercial product
- Brakeman may be used to scan any application, even as part of a consulting contract or professional security assessment, regardless of the commercial nature of the application
- All distributed modifications must be under the same terms as the original code
- The copyright for contributions made to https://github.com/presidentbeef/brakeman/ will be assigned to Synopsys, Inc.
